packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeSingleNode.tsx

```diff
@@ -11,121 +11,30 @@ import {
 import { darken, lighten } from 'polished';
 import { FC } from 'react';
 import { useToggle } from 'react-use';
 import { getItemBgColor } from '../../../../../hooks/useColumns';
 import { useFilters } from '../../../../../hooks/useFilters';
-import { rehypeRemoveHeaderLinks } from '../../../../../utils/markdownUtils';
 import FieldIcon from '../../../../common/Filters/FieldIcon';
 import MantineIcon from '../../../../common/MantineIcon';
 import { useItemDetail } from '../ItemDetailContext';
+import { ItemDetailMarkdown, ItemDetailPreview } from '../ItemDetailPreview';
 import { Node, useTableTreeContext } from './TreeProvider';
 import TreeSingleNodeActions from './TreeSingleNodeActions';
 
 type Props = {
     node: Node;
 };
 
-/**
- * Renders markdown for an item's description, with additional constraints
- * to avoid markdown styling from completely throwing its surroundings out
- * of whack.
- */
-const NodeDetailMarkdown: FC<{ source: string }> = ({ source }) => {
-    const theme = useMantineTheme();
-    return (
-        <MarkdownPreview
-            skipHtml
-            linkTarget="_blank"
-            components={{
-                h1: ({ children }) => <Title order={2}>{children}</Title>,
-                h2: ({ children }) => <Title order={3}>{children}</Title>,
-                h3: ({ children }) => <Title order={4}>{children}</Title>,
-            }}
-            rehypeRewrite={rehypeRemoveHeaderLinks}
-            source={source}
-            disallowedElements={['img']}
-            style={{
-                fontSize: theme.fontSizes.sm,
-            }}
-        />
-    );
-};
-
-/**
- * Renders a truncated version of an item's description, with an option
- * to read the full description if necessary.
- */
-const NodeDetailPreview: FC<{
-    description?: string;
-    onViewDescription: () => void;
-}> = ({ description, onViewDescription }) => {
-    if (!description) return null;
-
-    /**
-     * This value is pretty arbitrary - it's an amount of characters that will exceed
-     * a single line, and for which the 'Read more' option should make sense, and not
-     * be an annoyance.
-     *
-     * It's better to err on the side of caution, and show the 'Read more' option even
-     * if unnecessarily so.
-     */
-    const isTruncated = description.length > 180;
-
-    return (
-        <Flex direction="column" gap={'xs'}>
-            <Box
-                mah={140}
-                style={{
-                    overflow: 'hidden',
-                    textOverflow: 'clip',
-                }}
-            >
-                <NodeDetailMarkdown source={description} />
-            </Box>
-            {isTruncated && (
-                <Box
-                    ta={'center'}
-                    /**
-                     * Forces the 'Read more' option to slightly overlap with the content, and show
-                     * a slight fade effect.
-                     */
-                    mt={-30}
```

Comment:
To avoid hardcoded values and negative margin-top, we could change this div to absolute positioning and make the popover relative. Then, you can simply use bottom={0}